pkg/abaputils/abaputils.go

```diff
@@ -142,10 +142,13 @@ func HandleHTTPError(resp *http.Response, err error, message string, connectionD
 		var abapErrorResponse AbapError
 		bodyText, readError := ioutil.ReadAll(resp.Body)
 		if readError != nil {
-			return readError
+			return err
```

Comment:
I don't understand this. What value does "err" have here? readError contains the error. It is likely puzzling without context, so you could wrap it:

```suggestion
			return fmt.Errorf("failed to read response body: %w", readError)
```

Reply:
It's the origianl error passed to the function. If it's not possible to enrich this error with values from the response body, the original error should be returned